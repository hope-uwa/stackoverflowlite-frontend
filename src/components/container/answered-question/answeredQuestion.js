/* eslint-disable max-len */
/* eslint-disable class-methods-use-this */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loadAllQuestion } from '../../../actions/allQuestions/allQuestionAction';
import AllQuestion from '../../presentation/allquestions/AllQuestions';
import Banner from '../banner/Banner';

export class AnsweredQuestion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      answered: []
    };
  }

  componentDidMount =() => {
    this.props.loadQuestion();
  }

  componentWillReceiveProps(props) {
    const { allQuestions } = props;
    const answered = allQuestions.filter(x => x.count > 0);
    this.setState({ answered });
  }


  viewQuestion = (questionId) => {
    const {
      // eslint-disable-next-line react/prop-types
      history
    } = this.props;
    history.push(`/questions/${questionId}`);
  }

  render() {
    const { answered } = this.state;

    return (
      <div>
        <Banner title={'Answered Questions'}/>
        <div className=" allquestion row pl2 pls1">
          {answered.map((x, key) => <AllQuestion questions={x} key={key} view={this.viewQuestion} />)}

        </div>
        <div className="pagination">
          <a href="#">❮</a>
          <a href="#">❯</a>
        </div>
      </div>

    );
  }
}

AnsweredQuestion.propTypes = {
  loadQuestion: PropTypes.func,
  allQuestions: PropTypes.array,
};

export const mapStateToProps = (state) => {
  return {
    allQuestions: state.allQuestionReducer.questions,
    user: state.loginReducer
  };
};

export const mapDispatchToProps = {
  loadQuestion: () => loadAllQuestion()
};

export default connect(mapStateToProps, mapDispatchToProps)(AnsweredQuestion);
