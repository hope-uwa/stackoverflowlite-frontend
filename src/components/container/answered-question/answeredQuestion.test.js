import React from 'react';
import { shallow } from 'enzyme';
import { AnsweredQuestion, mapDispatchToProps, mapStateToProps } from './answeredQuestion';

describe('landing page', () => {
  const getProps = () => (
    {
      allQuestions: [
        {
          question_title: '',
          user_name: '',
        }
      ],

      loadQuestion: () => {},
      history: [],
    }
  );
  it('snap shot', () => {
    const props = getProps(true);
    const component = shallow(<AnsweredQuestion {...props}/>);
    expect(component).toMatchSnapshot();
  });

  it('should load viewQuestion', () => {
    const props = getProps();
    const component = shallow(<AnsweredQuestion {...props}/>);
    component.instance().viewQuestion('2');
    expect(component.instance().viewQuestion).toBeDefined();
  });
  it('should load componentWillRecieveProps', () => {
    const props = getProps();
    const component = shallow(<AnsweredQuestion {...props}/>);
    component.instance().componentWillReceiveProps({ ...props });
    expect(component.instance().componentWillReceiveProps).toBeDefined();
  });

  it('define mapStateToProps', () => {
    const state = { allQuestionReducer: { questions: [] } };
    expect(mapStateToProps(state)).toBeDefined();
  });
  it('define mapDispatchToProps', () => {
    expect(mapDispatchToProps).toBeDefined();
    expect(mapDispatchToProps.loadQuestion()).toBeDefined();
  });
});
