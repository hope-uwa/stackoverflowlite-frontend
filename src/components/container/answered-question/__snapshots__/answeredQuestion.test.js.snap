// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing page snap shot 1`] = `
<div>
  <Banner
    title="Answered Questions"
  />
  <div
    className=" allquestion row pl2 pls1"
  />
  <div
    className="pagination"
  >
    <a
      href="#"
    >
      ❮
    </a>
    <a
      href="#"
    >
      ❯
    </a>
  </div>
</div>
`;
